src/Npgsql/TypeHandlers/HstoreHandler.cs

```diff
@@ -30,6 +30,7 @@ namespace Npgsql.TypeHandlers
         typeof(Dictionary<string, string?>),
         typeof(IDictionary<string, string?>),
 #if !NET461 && !NETSTANDARD2_0 && !NETSTANDARD2_1
+        typeof(IReadOnlyDictionary<string, string?>),
```

Comment:
`IReadOnlyDictionary<,>` exists in all frameworks, so no need to put it inside the region.

Reply:
I put this here because `Read` for `IReadOnlyDictionary` is implemented using `ImmutableDictionary` which is not available in all frameworks. There are a few options here as I can see:

- Keep it as-is with read-only support available only in .NET Core 3.0.
- Switch the implementation for all frameworks over to `Dictionary` and include it everywhere.
- Conditionally implement it with `ImmutableDictionary` or `Dictionary` depending on what framework we're compiling for.

Which do you think is the best option?